adc.cpp

```diff
@@ -81,33 +79,9 @@ bool AdcExternal::init(I2C_Handle i2cHandle, uint8_t i2cAddress) {
     return true;
 }
 
-// Performs I2C transfer with txBuffer, rxBuffer, txCount, and rxCount
-// Returns true on transfer success, false on failure
-bool AdcExternal::transfer() {
-    bool ret;
-
-    /* Before performing any i2c transfer, verify we called init() first*/
-    if (i2cHandle == NULL) {
-        return false;
-    }
-
-    i2cTransaction.writeBuf = txBuffer;
-    i2cTransaction.writeCount = txCount;
-    i2cTransaction.readBuf = rxBuffer;
-    i2cTransaction.readCount = rxCount;
-
-    ret = I2C_transfer(i2cHandle, &i2cTransaction);
-
-    if (!ret) {
-        i2cErrorHandler(&i2cTransaction);
-    }
-
-    return ret;
-}
-
 // Sends uint8_t data to uint8_t reg
 // Returns true on transfer success, false on failure
-bool AdcExternal::setRawRegisterValue(Register reg, uint8_t data) {
+bool AdcExternal::setRawRegisterValue(uint8_t reg, uint8_t data) {

```

Comment:
How do we perform an I2C transaction if you remove `transfer()` from the class?

Reply:
Transfer isn't removed from the class, it is just the same code higher up right? If anything inheriting I2C needs a specific implementation then they can override it.